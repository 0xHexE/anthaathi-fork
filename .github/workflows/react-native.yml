name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
