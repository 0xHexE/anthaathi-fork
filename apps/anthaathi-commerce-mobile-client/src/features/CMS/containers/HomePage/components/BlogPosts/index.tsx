import {Image, Pressable, View} from 'react-native';
import React from 'react';
import {Divider, Text, TouchableRipple} from 'react-native-paper';
import {useIntl} from 'react-intl';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {useDimension} from '../../../../utils/useDimension';
import {HomePageComponentType} from '../../../../types/common';

type BlogsProps = {
  id: number;
  title: string;
  image: string;
  published_date: string;
  author: string;
};

export interface BlogPostProps {
  title: string;
  handlePress?: () => void;
  blogs: BlogsProps[];
  mainBlog: BlogsProps;
}

const BlogPosts = (props: BlogPostProps) => {
  const intl = useIntl();
  const itemHeight = useResponsiveValue([120, 120, 180, 240]);
  const itemMainHeight = useResponsiveValue([240, 320, 420, 480]);
  const [getWidth] = useDimension();

  return (
    <View style={{marginHorizontal: 10}} testID="blogPosts">
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}>
        <Text variant="titleLarge" style={{marginBottom: 9, fontSize: 20}}>
          {props.title}
        </Text>

        <Pressable onPress={props.handlePress} testID="handlePressBlogs">
          <Text
            variant="titleMedium"
            style={{
              marginBottom: 9,
              textDecorationLine: 'underline',
              fontSize: 14,
            }}>
            {intl.formatMessage({defaultMessage: 'View All'})}
          </Text>
        </Pressable>
      </View>

      <View style={{flexDirection: getWidth < 800 ? 'column' : 'row'}}>
        <View style={{width: getWidth < 800 ? '100%' : '40%'}}>
          <MainBlog blog={props.mainBlog} itemHeight={itemMainHeight} />
        </View>
        <Divider style={{marginVertical: 5}} />
        <View
          style={{
            width: getWidth < 800 ? '100%' : '58%',
            marginHorizontal: getWidth < 800 ? 0 : 10,
          }}>
          {props.blogs.map(blog => {
            return (
              <BlogRenderer key={blog.id} blog={blog} itemHeight={itemHeight} />
            );
          })}
        </View>
      </View>
    </View>
  );
};

const MainBlog = ({
  blog,
  itemHeight,
}: {
  blog: BlogsProps;
  itemHeight: number;
}) => {
  return (
    <TouchableRipple onPress={() => {}}>
      <View style={{alignItems: 'center', marginBottom: 10}}>
        <Image
          testID="mainBlogImage"
          source={{uri: blog.image}}
          style={{height: itemHeight, width: '100%'}}
        />
        <Text
          testID="mainBlogTitle"
          style={{
            fontSize: 16,
            fontWeight: '700',
            width: '70%',
            textAlign: 'center',
          }}>
          {blog.title}
        </Text>
        <Text testID="mainBlogDate" style={{fontSize: 14, fontWeight: '400'}}>
          {blog.published_date}
        </Text>
      </View>
    </TouchableRipple>
  );
};

const BlogRenderer = ({
  blog,
  itemHeight,
}: {
  blog: BlogsProps;
  itemHeight: number;
}) => {
  return (
    <TouchableRipple onPress={() => {}}>
      <View style={{flexDirection: 'row', marginBottom: 10}}>
        <Image
          source={{uri: blog.image}}
          style={{height: itemHeight, width: '40%'}}
        />
        <View style={{marginHorizontal: '2%', width: '56%'}}>
          <Text style={{fontSize: 16, fontWeight: '700'}}>{blog.title}</Text>
          <Text style={{fontSize: 14, fontWeight: '400'}}>
            {blog.published_date}
          </Text>
        </View>
      </View>
    </TouchableRipple>
  );
};

export default BlogPosts;

export const BlogPostsCMSInput = {
  _component: HomePageComponentType.BlogPosts,
  component: BlogPosts,
};
