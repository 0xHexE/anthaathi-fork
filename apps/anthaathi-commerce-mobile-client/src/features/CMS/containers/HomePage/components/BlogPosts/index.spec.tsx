import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import BlogPosts from './index';

describe('BlogPosts', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BlogPosts
            title="From the journal"
            mainBlog={{
              id: 1,
              title: 'blog title',
              image:
                'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
              published_date: 'May 26, 2022',
              author: 'author name',
            }}
            blogs={[
              {
                id: 2,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
              {
                id: 3,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('blogPosts')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BlogPosts
            title="From the journal"
            handlePress={onpress}
            mainBlog={{
              id: 1,
              title: 'blog title',
              image:
                'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
              published_date: 'May 26, 2022',
              author: 'author name',
            }}
            blogs={[
              {
                id: 2,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    fireEvent.press(temp.queryByTestId('handlePressBlogs')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have BlogPosts title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BlogPosts
            title="From the journal"
            mainBlog={{
              id: 1,
              title: 'blog title',
              image:
                'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
              published_date: 'May 26, 2022',
              author: 'author name',
            }}
            blogs={[
              {
                id: 2,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('From the journal')).toBeTruthy();
  });

  it('should have BlogPosts Main blog Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BlogPosts
            title="From the journal"
            mainBlog={{
              id: 1,
              title: 'blog title',
              image:
                'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
              published_date: 'May 26, 2022',
              author: 'author name',
            }}
            blogs={[
              {
                id: 2,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('mainBlogImage')!.props.source).toMatchObject({
      uri: 'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
    });
  });

  it('should have BlogPosts title, published_date', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BlogPosts
            title="From the journal"
            mainBlog={{
              id: 1,
              title: 'main blog title',
              image:
                'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg',
              published_date: 'May 26, 2022',
              author: 'author name',
            }}
            blogs={[
              {
                id: 2,
                title: 'blog title',
                image:
                  'https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg',
                published_date: 'May 26, 2022',
                author: 'author name',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('mainBlogTitle')?.children).toContain(
      'main blog title',
    );
    expect(temp.queryByTestId('mainBlogDate')?.children).toContain(
      'May 26, 2022',
    );
  });
});
