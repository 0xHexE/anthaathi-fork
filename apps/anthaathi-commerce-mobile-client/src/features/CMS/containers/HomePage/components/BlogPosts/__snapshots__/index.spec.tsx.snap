// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPosts should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="blogPosts"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "fontSize": 20,
            "marginBottom": 9,
          },
        ]
      }
    >
      From the journal
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="handlePressBlogs"
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 16,
              "fontWeight": "500",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "fontSize": 14,
              "marginBottom": 9,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        View All
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/Fathers-day-recipe-ideas-by-fresh-fruit-company-in-Dubai-1200x600_360x.jpg",
              }
            }
            style={
              Object {
                "height": 320,
                "width": "100%",
              }
            }
            testID="mainBlogImage"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "fontSize": 16,
                  "fontWeight": "700",
                  "textAlign": "center",
                  "width": "70%",
                },
              ]
            }
            testID="mainBlogTitle"
          >
            blog title
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "400",
                },
              ]
            }
            testID="mainBlogDate"
          >
            May 26, 2022
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "height": 0.5,
          },
          undefined,
          false,
          false,
          Object {
            "marginVertical": 5,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "marginHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg",
              }
            }
            style={
              Object {
                "height": 120,
                "width": "40%",
              }
            }
          />
          <View
            style={
              Object {
                "marginHorizontal": "2%",
                "width": "56%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "700",
                  },
                ]
              }
            >
              blog title
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              May 26, 2022
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/articles/5-ways-to-reduce-food-wastage-with-Fresh-fruits-and-Vegetables-1200x600_520x500.jpg",
              }
            }
            style={
              Object {
                "height": 120,
                "width": "40%",
              }
            }
          />
          <View
            style={
              Object {
                "marginHorizontal": "2%",
                "width": "56%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "700",
                  },
                ]
              }
            >
              blog title
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              May 26, 2022
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
